api/observability/v1/input_types.go

```diff
@@ -0,0 +1,294 @@
+type Infrastructure struct {
+	// Sources defines the list of infrastructure sources to collect.
+	// This field is optional and omission results in the collection of all infrastructure sources.
+	//
+	// +kubebuilder:validation:Optional
+	Sources []InfrastructureSource `json:"sources,omitempty"`
+}
+
+// AuditSource defines which type of audit log source is used.
+//
+// +kubebuilder:validation:Enum:=auditd;kubeAPI;openshiftAPI;ovn
+type AuditSource string
+
+const (
+	// AuditSourceKube are audit logs from kubernetes API servers
+	AuditSourceKube AuditSource = "kubeAPI"
+
+	// AuditSourceOpenShift are audit logs from OpenShift API servers
+	AuditSourceOpenShift AuditSource = "openshiftAPI"
+
+	// AuditSourceAuditd are audit logs from a node auditd service
+	AuditSourceAuditd AuditSource = "auditd"
+
+	// AuditSourceOVN are audit logs from an Open Virtual Network service
+	AuditSourceOVN AuditSource = "ovn"
+)
+
+var (
+	AuditSources = []AuditSource{
+		AuditSourceKube,
+		AuditSourceOpenShift,
+		AuditSourceAuditd,
+		AuditSourceOVN,
+	}
+)
+
+// Audit enables audit logs.
+type Audit struct {
+	// Sources defines the list of audit sources to collect.
+	// This field is optional and its exclusion results in the collection of all audit sources.
+	//
+	// +kubebuilder:validation:Optional
+	Sources []AuditSource `json:"sources,omitempty"`
+}
+
+// ReceiverType specifies the type of receiver that should be created.
+//
+// +kubebuilder:validation:Enum:=http;syslog
+type ReceiverType string
+
+const (
+	ReceiverTypeHttp   ReceiverType = "http"
+	ReceiverTypeSyslog ReceiverType = "syslog"
+)
+
+var (
+	ReceiverTypes = []ReceiverType{
+		ReceiverTypeHttp,
+		ReceiverTypeSyslog,
+	}
+)
+
+// ReceiverSpec is a union of input Receiver types.
+type ReceiverSpec struct {
+	// Type of Receiver plugin.
+	//
+	// +kubebuilder:validation:Required
+	Type ReceiverType `json:"type"`
+
+	// TLS contains settings for controlling options of TLS connections.
+	//
+	// +kubebuilder:validation:Optional
+	TLS *InputTLSSpec `json:"tls,omitempty"`
+
+	// Port the Receiver listens on. It must be a value between 1024 and 65535
+	//
+	// +kubebuilder:default:=8443
+	// +kubebuilder:validation:Minimum:=1024
+	// +kubebuilder:validation:Maximum:=65535
+	Port int32 `json:"port"`
+
+	// +kubebuilder:validation:Optional
+	HTTP *HTTPReceiver `json:"http,omitempty"`
+}
+
+// HTTPReceiverFormat defines the type of log data incoming through the HTTP receiver.
+//
+// +kubebuilder:validation:Enum:=kubeAPIAudit
+type HTTPReceiverFormat string
+
+const (
+	HTTPReceiverFormatKubeAPIAudit HTTPReceiverFormat = "kubeAPIAudit"
+)
+
+// HTTPReceiver receives encoded logs as a HTTP endpoint.
+type HTTPReceiver struct {
+	// Format is the format of incoming log data.
+	//
+	// +kubebuilder:validation:Required
+	Format HTTPReceiverFormat `json:"format"`
+}
+
+// InputTLSSpec contains options for TLS connections that are agnostic to the input type.
+type InputTLSSpec struct {
```

Comment:
I expect some of these fields are not compatible to be set together, would be helpful for end users to explain that within the docs